folium/map.py

```diff
@@ -297,6 +296,27 @@ def render(self, **kwargs):
             '</style>'), name='map_style')
 
         super(LegacyMap, self).render(**kwargs)
+    
+    @property
+    def location(self):
+        return self._location
+    
+    @location.setter
+    def location(self, value):
+        """Validates the location values before setting"""
+        if type(value) not in [list, tuple]:
+            raise TypeError("Location is not a list, expecting ex: location=[45.523, -122.675]")
+        
+        if len(value) != 2:
+            raise ValueError("Location should have two values, [lat, lon]")
+        
+        for val in value:
+            try:
+                float(val)
+            except:
+                raise TypeError("Location values should be numeric, {} is not a number".format(val))
```

Comment:
oh that's an interesting point. I guess I could actually convert the values to float or number and then we wouldn't need to know whether it works in leaflet or not.